File: 5/594/LongestHarmoniousSubsequence.java
/*
https://leetcode.com/problems/longest-harmonious-subsequence/description/?envType=daily-question&envId=2025-06-30
*/

class Solution {
    public int findLHS(int[] nums) {
        Arrays.sort(nums);
        int lengthOfNums=nums.length, resultantSubseqLength=0;
        for(int i=0; i<lengthOfNums; ++i){
            ArrayList<Integer>currentHarmonList=new ArrayList<>();
            for(int j=i; j<lengthOfNums; ++j){
                if(currentHarmonList.isEmpty()){
                    currentHarmonList.add(nums[j]);
                } else{
                    int numberToCheck=nums[j], currentMaxNumber=currentHarmonList.get(currentHarmonList.size()-1);
                    if(numberToCheck<nums[0]){
                        if(currentMaxNumber-numberToCheck==1){
                            currentHarmonList.add(numberToCheck);
                        }
                    } else if(numberToCheck>currentMaxNumber){
                        if(numberToCheck-currentHarmonList.get(0)==1){
                            currentHarmonList.add(numberToCheck);
                        } else break;
                    } else currentHarmonList.add(numberToCheck);
                }
            }
            int harmonListSize=currentHarmonList.size();
            int currentMaxNumber=currentHarmonList.get(currentHarmonList.size()-1);
            if(currentMaxNumber-currentHarmonList.get(0)==1){
                resultantSubseqLength=Math.max(resultantSubseqLength, harmonListSize);
            }
        }
        return resultantSubseqLength;
    }
}

File: 5/594/LongestHarmoniousSubsequence.cpp
/*
https://leetcode.com/problems/longest-harmonious-subsequence/description/?envType=daily-question&envId=2025-06-30
*/

class Solution {
public:
    int findLHS(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int sz=nums.size(), res=0;
        for(int i=0; i<sz; ++i){
            vector<int>curr;
            for(int j=i; j<sz; ++j){
                if(curr.size()==0){
                    curr.push_back(nums[j]);
                } else{
                    int tochk=nums[j];
                    if(tochk<curr[0]){
                        if(curr[curr.size()-1]-tochk==1){
                            curr.push_back(nums[j]);
                        }
                    } else if(tochk>curr[curr.size()-1]){
                        if(tochk-curr[0]==1){
                            curr.push_back(nums[j]);
                        } else break;
                    } else{
                        curr.push_back(nums[j]);
                    }
                }
            }
            int tohcksz=curr.size();
            if(curr[curr.size()-1]-curr[0]==1)
                res=max(res, tohcksz);
        }
        return res;
    }
};